packages/web3-shh/types/index.d.ts

```diff
@@ -0,0 +1,130 @@
+    (at your option) any later version.
+    web3.js is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Lesser General Public License for more details.
+    You should have received a copy of the GNU Lesser General Public License
+    along with web3.js.  If not, see <http://www.gnu.org/licenses/>.
+*/
+/**
+ * @file index.d.ts
+ * @author Josh Stevens <joshstevens19@hotmail.co.uk>, Samuel Furter <samuel@ethereum.org>
+ * @date 2018
+ */
+
+import * as net from 'net';
+import { AbstractWeb3Module, provider, Web3ModuleOptions } from 'web3-core';
+import { Network } from 'web3-net';
+
+export class Shh extends AbstractWeb3Module {
+    constructor(
+        provider: provider,
+        net?: net.Socket|null,
+        options?: Web3ModuleOptions
+    );
+
+    net: Network;
+
+    getVersion(callback?: (error: Error, version: string) => void): Promise<string>;
+
+    getInfo(callback?: (error: Error, info: Info) => void): Promise<Info>;
+
+    setMaxMessageSize(size: number, callback?: (error: Error, result: boolean) => void): Promise<boolean>;
+
+    setMinPoW(pow: number, callback?: (error: Error, result: boolean) => void): Promise<boolean>;
+
+    markTrustedPeer(enode: string, callback?: (error: Error, result: boolean) => void): Promise<boolean>
+
+    newKeyPair(callback?: (error: Error, key: string) => void): Promise<string>;
+
+    addPrivateKey(privateKey: string, callback?: (error: Error, privateKey: string) => void): Promise<string>;
+
+    deleteKeyPair(id: string, callback?: (error: Error, result: boolean) => void): Promise<boolean>;
+
+    hasKeyPair(id: string, callback?: (error: Error, result: boolean) => void): Promise<boolean>;
+
+    getPublicKey(id: string, callback?: (error: Error, publicKey: string) => void): Promise<string>;
+
+    getPrivateKey(id: string, callback?: (error: Error, privateKey: string) => void): Promise<string>;
+
+    newSymKey(callback?: (error: Error, key: string) => void): Promise<string>;
+
+    addSymKey(symKey: string, callback?: (error: Error, key: string) => void): Promise<string>;
+
+    generateSymKeyFromPassword(password: string, callback?: (error: Error, key: string) => void): Promise<string>;
+
+    hasSymKey(id: string, callback?: (error: Error, result: boolean) => void): Promise<boolean>;
+
+    getSymKey(id: string, callback?: (error: Error, key: string) => void): Promise<string>;
+
+    deleteSymKey(id: string, callback?: (error: Error, result: boolean) => void): Promise<boolean>
+
+    post(object: PostWithSymKey | PostWithPubKey, callback?: (error: Error, result: string) => void): Promise<string>;
+
+    subscribe(string: 'messages', options: SubscriptionOptions, callback?: (error: Error, message: Notification, subscription: any) => void): Subscribe;
+
+    newMessageFilter(options?: SubscriptionOptions): Promise<string>;
```

Comment:
missing callback parameter for newMessageFilter, deleteMessageFilter, and getFilterMessages.